### DISCUSSION TITLE
Concerns about GPT-4 Fees for Students

### DISCUSSION THREAD

**Discussion by williamstein** [sagemathinc/cocalc#6879] [Q&A]
CoCalc now provides GPT-4 on a pay-for-what-you-use basis, in addition to our free GPT-3.5 functionality.  As an instructor or student, you might have some questions about how this works!

> 1) I don't see a warning about fees when I select "@GPT-4" in the chat window. Does the platform remind users about the fee before the chat is sent?

Anybody can select GPT-4 (in chat and other places), but the first time you use it, there is a big confirmation dialog.  This lets you set a specific monthly spending limit (you can set anything you want), which is by default $0.    You can always adjust this limit later at https://cocalc.com/settings/purchases under "Self-Imposed Spending Limits", where you can also see the rates.

The dialog also lets you add credit to your account, in case you don't have any, and you can check on the status of that credit at https://cocalc.com/settings/purchases.   After you explicitly set a limit and add credit, you don't get explicitly asked again every time you use GPT-4.  Also, on any day when you use GPT-4, *you'll receive an email statement at the end of the day listing how much you spent* (and this is easy to disable).
 
> 2) If a student uses GPT-4 once, will CoCalc default to GPT-4 thereafter?

Currently no.  The default is always GPT-3.5.  That said, several people have been requesting a way to default to GPT-4, to save themselves a click, so we will very likely make that an option sometime in the near future.  But it will be possible to configure it either way.

> From the "tokens" pricing scheme on OpenAI's site, it is difficult for me to get a good approximation for how much GPT-4 use would cost my students. I recognize that there are too many unknowns for a specific dollar amount, but can you give me any information that would help estimate the cost per semester? A sense of scale ($1 vs. $10 vs. $100 per semester) would be helpful.

Since all use is explicit and manual, e.g., via chat or clicking, in practice it's very difficult to use very much.    My guess is that a typical student **might use $10 for an entire semester worth of use**.     A typical interaction is a few cents, so hundreds of interactions cost about $10.     You'll quickly get a sense of spend because it's listed in the daily statements mentioned above.   For comparison, OpenAI charges $20/month for their GPT-4 chat site, and Microsoft charges $30/month for their CoPilot integration. The model in cocalc where you pay for what you actually use is more affordable.
 
Note that GPT-3.5 is significantly faster (and completely free to users, though it costs me), and for some things it's pretty good, so people often use it just because the output appears so quickly. 
 
Some other notes:

- In case you're worried, it's also possible to fully or partly disable ChatGPT for students in your class, e.g., during an exam. That's in course configuration.

- We're planning to add other Large Language Models, e.g. Claude2 from Anthropic, pretty soon.

**Comment by walishah** [comment]
hi

**Reply by Teja-2006** [reply]
Ola
